src/transformers/tokenization_utils_fast.py

```diff
@@ -870,3 +877,53 @@ def train_new_from_iterator(
             kwargs["additional_special_tokens"] = additional_special_tokens
 
         return self.__class__(tokenizer_object=tokenizer, **kwargs)
+
+    def update_post_processor(self):
+        """
+        Updates the underlying post processor with the current `bos_token` and `eos_token`.
+        """
+        if not isinstance(self._tokenizer.post_processor, processors.TemplateProcessing) and not isinstance(
+            self._tokenizer.post_processor, processors.Sequence
+        ):
+            return
+
+        bos = self.bos_token
+        bos_token_id = self.bos_token_id
+        if bos is None and self.add_bos_token:
+            raise ValueError("add_bos_token = True but bos_token = None")
+
+        eos = self.eos_token
+        eos_token_id = self.eos_token_id
+        if eos is None and self.add_eos_token:
+            raise ValueError("add_eos_token = True but eos_token = None")
+
+        single = f"{(bos + ':0 ') if self.add_bos_token else ''}$A:0{(' ' + eos + ':0') if self.add_eos_token else ''}"
+        pair = f"{single}{(' ' + bos + ':1') if self.add_bos_token else ''} $B:1{(' ' + eos + ':1') if self.add_eos_token else ''}"
+
+        special_tokens = []
+        if self.add_bos_token:
+            special_tokens.append((bos, bos_token_id))
+        if self.add_eos_token:
+            special_tokens.append((eos, eos_token_id))
+        if special_tokens:
+            self._tokenizer.post_processor = processors.TemplateProcessing(
+                single=single, pair=pair, special_tokens=special_tokens
+            )
```

Comment:
We could add a warning: if `self._tokenizer.post_processor` is not None, then "Warning overwriting the original postProcessor. Reload the tokenizer if that is not desired" WDYT?

Reply:
Sounds good! Should we add that it is due to passing in bos/eos token?

ex. "Warning overwriting the original postProcessor in order to update `bos_token` or `eos_token`. Reload the tokenizer without these parameters if that is not desired"

Reply:
Yeah sounds good

Reply:
Why is `self._tokenizer.post_processor` only updated if `special_tokens` is non-empty? If we toggle BOS and EOS off, shouldn't the processor get updated?